force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,34 +259,43 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
-
+        if (d1 == null || d2 == null) {
+            return null;
+        }
         // Subtract the difference in years
+        Integer y1 = d1.year();
+        Integer y2 = d2.year();
+        Integer yearDiff = Math.abs(y1 - y2);
 
-        return null; // Replace null with the variable you used to store the result
+        return yearDiff; // Replace null with the variable you used to store the result
```

Comment:
Logic is fine. I think adding variables like y1 and y2 makes it a little harder to read year1 or year2 would have been better.  But you don't need them 

```
  public static Integer calculateYearDifference(Date d1, Date d2) {
        if (d1 == null || d2 == null) {
            return null;
        }
        return Math.abs(d1.year() - d2.year());
    }
    ```